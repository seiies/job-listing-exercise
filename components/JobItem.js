import Link from 'next/link'
import Paper from 'material-ui/Paper'
import PropTypes from 'prop-types'
import React from 'react'
import config from '../next.config.js'

const JobItem = props => {
  const {
    job: {
      id,
      image,
      company,
      title
    }
  } = props

  const jobPageConfig = config.routes.jobPage
  const jobUrl = {
    as: jobPageConfig.routeWithParam.replace(':id', id),
    href: {
      pathname: jobPageConfig.route,
      query: { id: id }
    }
  }

  return (
    <React.Fragment>
      { /* language=CSS */ }
      <style jsx>{`
          .job {
            margin-bottom: 1em;
          }

          .job:last-child {
            margin-bottom: 0;
          }

          .block {
            display: flex;
            align-items: flex-start;
            padding: 1rem;
          }

          .blockImg {
            width: 6.25rem;
            margin: 0 1rem 0 0;
          }

          .blockBody {
            flex: 1;
          }

          a {
            text-decoration: none;
          }

          .jobTitle {
            font-size: 1.5rem;
            margin-top: 0;
            margin-bottom: .5rem;
            color: black;
          }

          .companyName {
            margin: 0;
            font-size: .9rem;
            color: gray;
          }
        `}</style>

      <li className='job'>
        <Link href={jobUrl.href} as={jobUrl.as}>
          <a title='Job full description'>
            <Paper elevation={3}>
              <div className='block'>
                <img className='blockImg' src={image} alt='Company logo' />
                <div className='blockBody'>
                  <h2 className='jobTitle'>{title}</h2>
                  <p className='companyName'>{company}</p>
                </div>
              </div>
            </Paper>
          </a>
        </Link>
      </li>
    </React.Fragment>
  )
}

JobItem.propTypes = {
  className: PropTypes.string,
  job: PropTypes.shape({
    id: PropTypes.number,
    image: PropTypes.string,
    company: PropTypes.string,
    title: PropTypes.string
  }).isRequired
}

JobItem.defaultProps = {}

export default JobItem
