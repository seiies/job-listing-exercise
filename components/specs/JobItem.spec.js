/* global describe, it, expect */

import React from 'react'
import JobItem from '../JobItem'
import { shallow } from 'enzyme'

describe('JobItem', () => {
  let job = {
    'company': 'Deutsche Bahn AG',
    'id': 1,
    'image': 'https://place.cat/c/100/100',
    'title': 'Job title'
  }

  it('renders correctly', () => {
    const component = shallow(<JobItem job={job} />)
    expect(component).toMatchSnapshot()
  })
})
