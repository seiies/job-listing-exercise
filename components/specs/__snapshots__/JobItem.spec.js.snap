// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem renders correctly 1`] = `
<React.Fragment>
  <JSXStyle
    css=".job.jsx-3219546763{margin-bottom:1em;}.job.jsx-3219546763:last-child{margin-bottom:0;}.block.jsx-3219546763{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;padding:1rem;}.blockImg.jsx-3219546763{width:6.25rem;margin:0 1rem 0 0;}.blockBody.jsx-3219546763{-webkit-flex:1;-ms-flex:1;flex:1;}a.jsx-3219546763{-webkit-text-decoration:none;text-decoration:none;}.jobTitle.jsx-3219546763{font-size:1.5rem;margin-top:0;margin-bottom:.5rem;color:black;}.companyName.jsx-3219546763{margin:0;font-size:.9rem;color:gray;}"
    styleId="3219546763"
  />
  <li
    className="jsx-3219546763 job"
  >
    <Link
      as="/job/1"
      href={
        Object {
          "pathname": "/job",
          "query": Object {
            "id": 1,
          },
        }
      }
    >
      <a
        className="jsx-3219546763"
        title="Job full description"
      >
        <WithStyles(Paper)
          elevation={3}
        >
          <div
            className="jsx-3219546763 block"
          >
            <img
              alt="Company logo"
              className="jsx-3219546763 blockImg"
              src="https://place.cat/c/100/100"
            />
            <div
              className="jsx-3219546763 blockBody"
            >
              <h2
                className="jsx-3219546763 jobTitle"
              >
                Job title
              </h2>
              <p
                className="jsx-3219546763 companyName"
              >
                Deutsche Bahn AG
              </p>
            </div>
          </div>
        </WithStyles(Paper)>
      </a>
    </Link>
  </li>
</React.Fragment>
`;
